doc/release/technotes/README.libraries.rst

```diff
@@ -23,36 +26,141 @@ the ``chpl`` command line take precedence over any found through object
-named ``<myModule>``, the corresponding initialization routine is
-named ``chpl__init_<myModule>``.  Therefore, for example, a C program that
-makes use of the standard Chapel module ``Random`` should
-call ``chpl__init_Random`` at some point before the Random interface is first
-used.
+When creating a library file from Chapel code, only those symbols with
+``export`` attached to them will be available from outside the library.  For
+example:
+
+.. code-block:: Chapel
+
+   // This function will be available to the library user
+   export proc foo() {
+     // Does something
+     ...
+   }
+
+   // but this function will not be
+   proc bar() {
+     // Does something else
+     ...
+   }
+
+When using a Chapel library file in C code, a fairly exact incantation is
+required.  First, a header file must be created for the library.  This does not
+happen through use of the Chapel ``--library`` flag during compilation!  If
+during compilation of the library the ``--savec <dir>`` flag is also used, the C
+definitions for the exported functions should be visible in a C file under
+*dir* of the same name as the library Chapel file.  A declaration version of
+these definitions should go into a ``.h`` file.  For instance, if one compiled
+a Chapel file ``foo.chpl`` like this:
+
+.. code-block:: sh
+
+   chpl --library --savec cDir --static -o libfoo foo.chpl
+
+
+then under ``cDir/foo.c`` one might see something resembling:
 
 .. code-block:: C
 
-    // C code using the Chapel Random module.
-    int main()
-    {
-      _int64 seed = 0;
-      double rand;
+   int64_t bar(void) {
+     ...
+   }
+
+   void baz(int64_t x) {
+     ...
+   }
+
+from which one would create a ``.h`` file:
+
+.. code-block:: C
+
+   int64_t bar(void);
+   void baz(int64_t x);
+
+
+Next, update the ``$LD_LIBRARY_PATH`` environment variable to include the
+directory where the new library file lives and the directory where the Chapel
+build lives.  The latter can be found by looking at the output of a
+``$CHPL_HOME/util/printchplenv`` call and finding the appropriate directory
+under ``$CHPL_HOME/lib``; the directory name is of the form
+``$CHPL_HOST_PLATFORM.$CHPL_HOST_COMPILER.(<env variable suffix>-<corresponding environment variable.)+``.
+For instance, if the output of a printchplenv call was::
+
+   CHPL_HOST_PLATFORM: linux64
+   CHPL_HOST_COMPILER: gnu
+   CHPL_TARGET_PLATFORM: linux64
+   CHPL_TARGET_COMPILER: gnu
+   CHPL_TARGET_ARCH: native
+   CHPL_LOCALE_MODEL: flat
+   CHPL_COMM: none
+   CHPL_TASKS: fifo
+   CHPL_LAUNCHER: none
+   CHPL_TIMERS: generic
+   CHPL_MEM: cstdlib
+   CHPL_MAKE: gmake
+   CHPL_ATOMICS: intrinsics
+   CHPL_GMP: gmp
+   CHPL_HWLOC: none
+   CHPL_REGEXP: re2
+   CHPL_WIDE_POINTERS: struct
+   CHPL_LLVM: none
+   CHPL_AUX_FILESYS: none
+
+then the ``$CHPL_HOME/lib/`` subdirectory is::
+
+     linux64.gnu.arch-native.loc-flat.comm-none.tasks-fifo.tmr-generic.mem-cstdlib.atomics-intrinsics.gmp.hwloc-none.re2.wide-struct.fs-none/
+
```

Comment:
I'm sure there is a better way to figure out this directory...

You could just run
./util/printchplenv --runtime